src/main/java/executor/service/service/impl/StepExecutionSleepImpl.java

```diff
@@ -1,17 +1,40 @@
 package executor.service.service.impl;
 
+import executor.service.exceptions.StepExecutionException;
 import executor.service.model.Step;
+import executor.service.model.StepTypes;
 import executor.service.service.StepExecutionSleep;
+import java.util.Random;
 import org.openqa.selenium.WebDriver;
 
 public class StepExecutionSleepImpl implements StepExecutionSleep {
-    @Override
-    public String getStepAction() {
-        return null;
-    }
 
-    @Override
-    public void step(WebDriver webDriver, Step step) {
+  @Override
+  public String getStepAction() {
+    return StepTypes.SLEEP.getName();
+  }
 
+  @Override
+  public void step(WebDriver webDriver, Step step) {
+    try {
+      Thread.sleep(getRandom(step.getValue()));
+    } catch (InterruptedException e) {
+      throw new StepExecutionException(e);
     }
+
+  }
+
+  private int getRandom(String value) {
+
+    String values[] = value.split(":");
```

Comment:
You can use **String.indexOf** and **String.substring** to extract parts of a string more clearly without creating an array of strings.

Reply:
I think it more transparent for undestarnding to use split. Than separation from string.

Reply:
we'd better ask about time format)